### PR TITLE
Add Controller for TidbClusterAutoScale

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
#1651 
Add Controller for TidbClusterAutoScaler and make it controlled by feature gate.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/autoscaler/tidbcluster_autoscaler_control.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+type defaultAutoScalerControl struct {
+	recoder      record.EventRecorder
+	typedControl controller.TypedControlInterface
+	//	autoScalerManager
+}
+
+func (tac *defaultAutoScalerControl) ResconcileAutoScaler(ta *v1alpha1.TidbClusterAutoScaler) error {
+	var errs []error
```
Comment (DanielZhangQD) [inline pkg/controller/autoscaler/tidbcluster_autoscaler_control.go]: An array is necessary here?

Comment (Yisaer) [reply pkg/controller/autoscaler/tidbcluster_autoscaler_control.go]: Not necessary and not important.

File: charts/tidb-operator/values.yaml [hunk 2]
```diff
@@ -34,6 +34,7 @@ imagePullPolicy: IfNotPresent
 features: []
 # - AdvancedStatefulSet=false
 # - StableScheduling=true
+# - AutoScaling=true
```
Comment (cofyc) [inline charts/tidb-operator/values.yaml]: use default value `false`

File: pkg/features/features.go [hunk 3]
```diff
@@ -41,6 +41,9 @@ const (
 
 	// AdvancedStatefulSet controls whether to use AdvancedStatefulSet to manage pods
 	AdvancedStatefulSet string = "AdvancedStatefulSet"
+
+	// AutoScaling controls whether to use TidbClusterAutoScaler to auto scale-in/out pods
+	AutoScaling string = "AutoScaling"
```
Comment (cofyc) [inline pkg/features/features.go]: add `AutoScaling: false` here

https://github.com/pingcap/tidb-operator/blob/0624cfe4694533ea9dca682811a4e4b7856cfb06/pkg/features/features.go#L30-L33

Comment (Yisaer) [reply pkg/features/features.go]: updated.

File: charts/tidb-operator/values.yaml (hunk 2 above)
Comment (Yisaer) [reply charts/tidb-operator/values.yaml]: updated.

**Pull request merged**